src/test/java/com/gmail/service/UserServiceImplTest.java

```diff
@@ -0,0 +1,90 @@
+package com.gmail.service;
+
+import com.gmail.config.JavaConfig;
+import com.gmail.model.User;
+import junit.framework.TestCase;
+import org.junit.Before;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import javax.sql.DataSource;
+import java.time.LocalDate;
+import java.util.List;
+
+public class UserServiceImplTest extends TestCase {
```

Comment:
Please use mockito to mock repository level.

Reply:
Why?

Reply:
You run tests on production data it is not a good idea. One of the way to fix it to use mockito or create database especially for tests (for example h2 database).